gateway-server/src/main/java/org/apache/knox/gateway/services/factory/TokenStateServiceFactory.java

```diff
@@ -61,17 +62,32 @@ protected Service createService(GatewayServices gatewayServices, ServiceType ser
           service.init(gatewayConfig, options);
         } catch (ServiceLifecycleException e) {
           LOG.errorInitializingService(implementation, e.getMessage(), e);
-          service = new AliasBasedTokenStateService();
-          ((AliasBasedTokenStateService) service).setAliasService(getAliasService(gatewayServices));
+          service = useDerbyDatabaseTokenStateService(gatewayServices, gatewayConfig, options);
         }
+      } else if (matchesImplementation(implementation, DerbyDBTokenStateService.class, true)) {
```

Comment:
I'll make that change anyway and move the`DerbyDBTokenStateService` so that the list starts with the `default` implementation (naming is not our friend here :) ). I may also rename `DefaultTSS` to `InMemoryTSS`...

Reply:
There is something about this logic that smells bad to me. From my point of view, someone should be able to implement TokenStateService with whatever impl they need, configure it in gateway-site, and drop their JAR in the ext dir. With this logic, that won't work. Can we consider refactoring this to keep Knox extensible?

Reply:
@pzampino good point, service loader.

Reply:
The service loader approach is already in use by `org.apache.knox.gateway.services.GatewayServiceFactory`. Thus, if someone wants to add their own implementation without contributing here, in the Knox project, they can do that by overriding the existing `TokenStateServiceFactory` and having that custom factory inserted into the `org.apache.knox.gateway.services.ServiceFactory` file within their custom JAR. 

However, I see that's a different discussion we should start in Knox's DEV e-mail list. As of now, all service implementations work that way and I would rather not do this work in the scope of this JIRA.
Any objection?

Reply:
Let's create a JIRA improvement issue to address the extensibility conversation. With that, I agree we can exclude that from the scope of this specific work.